test/common/testlib.py

```diff
@@ -1022,7 +1022,8 @@ def enable_preload(self, package, *pages):
     def system_before(self, version):
         try:
             v = self.machine.execute("""rpm -q --qf '%{V}' cockpit-system ||
-                                        dpkg-query -W -f '${source:Upstream-Version}' cockpit-system
+                                        dpkg-query -W -f '${source:Upstream-Version}' cockpit-system ||
+                                        pacman -Q cockpit | awk '{ print $2 }'
```

Comment:
This is not quite right -- you need to strip off the distro version ("-1") as well. I independently did that this morning in https://github.com/cockpit-project/cockpit/pull/16662 as this is blocking image refreshes. After that lands, you can remove this commit.